src/adaptors/yjs-ndn-adaptor.ts

```diff
@@ -96,9 +115,117 @@ export class NdnSvsAdaptor {
     if (this.#bundler) {
       await this.#bundler.produce(content);
     } else {
-      await this.syncAgent.publishUpdate(this.topic, content);
+      // await this.syncAgent.publishUpdate(this.topic, content);
+
+      // Adam Chen Injection point 1 override
+      await this.publishUpdate(this.topic, content);
+    }
+  }
+
+  // Adam Chen Injection point 1
+  private async publishUpdate(topic: string, content: Uint8Array) {
+    await this.syncAgent.publishUpdate(topic, content);
+    // await new Promise(r => setTimeout(r,500));
+    // forced wait so that publishUpdate() is completed before we check SV.
+    console.log('-- Injection point 1: Check StateVector / Create Snapshot --');
+    const stateVector = this.syncAgent.getUpdateSyncSV();
+    console.log('debug: stateVector object: ', stateVector);
+    let count = 0;
+    for (const [_id, seq] of stateVector) {
+      count += seq;
+    }
+    console.log('Total count of state vector', count);
+    console.log('The above number should match the state vector in the debug page');
+    if (count % 5 == 0) {
+      console.log("It's time to make a snapshot!");
+      console.log('debug: group prefix: ', this.syncAgent.appPrefix.toString());
+      const encodedSV = Encoder.encode(stateVector);
+      const snapshotPrefix = this.syncAgent.appPrefix.append('32=snapshot');
+      // New SVS encodings
+      const snapshotName = snapshotPrefix.append(new Component(Version.type, encodedSV));
+      console.log('debug: targeted snapshot Prefix (persistStore key): ', snapshotPrefix.toString());
+      // /groupPrefix/32=snapshot/
+      console.log('debug: targeted snapshot Name: ', snapshotName.toString());
+      // /groupPrefix/32=snapshot/54=<stateVector>
+      const decodedSV = Decoder.decode(snapshotName.at(-1).value, StateVector);
+      console.log('debug: decoding encoded SV from snapshotName: ', decodedSV);
+      let count = 0;
+      for (const [_id, seq] of decodedSV) {
+        count += seq;
+      }
+      console.log('debug: decoding encoded SV total packet count: ', count);
+      console.log('This should match the state vector in the debug page and the previous count before encoding');
+
+      const content = Y.encodeStateAsUpdate(this.doc);
+      // its already in UTF8, transporting currently without any additional encoding.
```

Comment:
I don't think it is plain text (UTF-8). I guess you mean Uint8Array.

Reply:
Sorry was drunk while writing this. Corrected.